backend/database/Search.py

```diff
@@ -1,38 +1,121 @@
 import re
+from backend.database.models import query_single_column
 
-class Search:
 
-    def __init__(self, source_string):
-        self.source_string = source_string
-        self.results = None
+class Search:
 
     @staticmethod
-    def parse(source_string):
-        sql_string = """
-            SELECT points.point_id FROM points
-            LEFT JOIN points_tags pt ON points.point_id = pt.point_id
-            WHERE
+    def parse(source_string, search_type):
+        """Generates a SQL WHERE-clause from the given search source string.
+
+        :param source_string: The source of the search to be parsed
+        :param search_type: A string representing the table being searched; one of 'point',
+        'device', 'room' or 'building'
+        :return: The generated SQL WHERE-clause
         """
+        sql_string = " WHERE "
 
         # @ -> buildings, # -> tags, $ -> rooms, % -> devices, * -> points
-        regex = re.compile("([@$%*]\\d+|#\\w+|and|or|not|:(\\w+) (([<>=]|!=|<=|>=) (\\d+)|(\"[^\"]+\"))|\\(|\\))")
+        regex = re.compile(
+            "([@#$%*]\\d+|and|or|not|:(\\w+) (([<>=]|!=|<=|>=)? ?(\\d+)|(\'\\w+\'))|\\(|\\))")
         # TODO: think harder about the name token
         for token in regex.findall(source_string):
             token = token[0]
-            print(type(token))
```

Comment:
Should we switch these print statements to logs? If so, should we use some logging handler for python/is there something like that built into python?

Reply:
Oops i just realized these print statements were in the deleted part of the diff. The idea still stands though if we want to add logs elsewhere

Reply:
This is being logged by flask as the search portion of the url.

We are also printing point names to the standard error for points that aren't in the database when something tries to add values for them. This will be in a log file on the server.